packages/Telephony/Common/RegexAggregatorInput.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property.
+    /// </summary>
+    public class RegexAggregatorInput : Dialog
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Telephony.RegexAggregatorInput";
+        private const string AggregationDialogMemory = "dialog.aggregation";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RegexAggregatorInput"/> class.
+        /// </summary>
+        /// <param name="sourceFilePath">Optional, source file full path.</param>
+        /// <param name="sourceLineNumber">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public RegexAggregatorInput([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        /// <summary>
+        /// Gets or sets the regex pattern to use to decide when the dialog has aggregated the whole message.
+        /// </summary>
+        [JsonProperty("terminationConditionRegexPattern")]
+        public StringExpression TerminationConditionRegexPattern { get; set; }
+
+        /// <summary>
+        /// Gets or sets the property to assign the result to.
+        /// </summary>
+        [JsonProperty("property")]
+        public StringExpression Property { get; set; }
+
+        /// <summary>
+        /// Gets or sets the activity to send to the user.
+        /// </summary>
+        /// <value>
+        /// An activity template.
+        /// </value>
+        [JsonProperty("prompt")]
+        public ITemplate<Activity> Prompt { get; set; }
+
+        /// <inheritdoc/>
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            ITemplate<Activity> template = this.Prompt ?? throw new InvalidOperationException($"InputDialog is missing Prompt.");
```

Comment:
if we dont want to support this functionality for non-telephony channel then we can add this line if (dc.Context.Activity.ChannelId == Channels.Telephony)
            {